include/gridtools/fn/backend/naive/apply.hpp

```diff
@@ -63,6 +63,80 @@ namespace gridtools::fn {
                     std::apply(Stencil, tuple_util::transform(make_iterator(ptr, strides), in_tags_t()));
             })(ptr, strides);
         }
+
+        template <class Key>
+        struct at_generator {
+            auto const &operator()(auto const &m) const { return at_key<Key>(m); }
+        };
+
+        template <class Key, class Map>
+        auto remove_key(Map const &m) {
+            using res_t = hymap::from_meta_map<meta::mp_remove<hymap::to_meta_map<Map>, Key>>;
+            using generators_t = meta::transform<at_generator, get_keys<res_t>>;
+            return tuple_util::generate<generators_t, res_t>(m);
```

Comment:
Maybe it makes sense to extract a small algorithm, e.g. hymap::filter, because I implemented more or less the same for the int_vector::normalize.

Reply:
There is a problem here: Even though the implementation is tiny and the name (`hymap::filter`)  looks self explaining, the contract about this function can be defined in a several ways and it is not clear what is the right way:
 - About the compile time predicate: it can be defined as a metafunction that takes a `meta::list<Key, ValueType>` or a metafunction that takes a key and a value type separately. 
 -  What happens when we are not removing items? Should we return the input unchanged? Or we should convert it to `hymap::key<...>::values` type anyway?

Probably each time you want to reuse that function you need to go to `hymap.hpp` and to recall the contract. Taking into account that the use cases are rare and implementation is tiny and straightforward, for me it is not obvious that we should factor this algorithm out.